### PR TITLE
fix: add build preflight checks and tag-aware catalog delete

### PR DESCRIPTION
**Pull request by abaker-9**
Build preflight checks:
- Add RunPreflight domain logic in pkg/solution/builder/preflight.go with lint and functional test validation before builds
- Add --skip-lint, --skip-tests, --ignore-preflight, and --allow-dev-version flags to build solution command
- Block builds when metadata.version is unset unless --allow-dev-version is passed
- Skip preflight during --dry-run

Catalog remote delete improvements:
- Add tag-aware delete fallback in RemoteCatalog.Delete for registries that reject digest-based DELETE (e.g., GCP Artifact Registry "dangling tag" error)
- Add InferKindFromRemote to probe remote catalog when local kind inference fails
- Wire remote kind inference into catalog delete command as fallback when --kind is omitted and local catalog is unavailable
- Fix fake OCI registry DELETE handler to resolve tags to digests

### CODE REVIEW

File: pkg/cmd/scafctl/build/solution.go [hunk 1]
```diff
@@ -260,6 +276,35 @@ func runBuildSolution(ctx context.Context, opts *SolutionOptions) error {
...
+		reg, regErr := builtin.DefaultRegistry(ctx)
+		if regErr != nil {
+			reg = provider.GetGlobalRegistry()
+		}
+
+		pfResult, pfErr := builder.RunPreflight(ctx, &sol, opts.File, builder.PreflightOptions{
+			SkipLint:        opts.SkipLint,
+			SkipTests:       opts.SkipTests,
+			IgnorePreflight: opts.IgnorePreflight,
+			Registry:        reg,
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go:304]: Fixed — `RunPreflight` now receives `BinaryPath` via `os.Executable()` so functional test pre-flight checks actually execute. Thanks!

File: pkg/cmd/scafctl/catalog/delete.go [hunk 2]
```diff
@@ -249,6 +249,24 @@ func runDeleteRemote(ctx context.Context, opts *DeleteOptions) error {
 		return exitcode.WithCode(err, exitcode.CatalogError)
 	}
 
+	// If kind is still unknown (short-reference path where ParseReference was
+	// skipped because no --kind was provided and local inference failed),
+	// infer from the remote catalog by probing each artifact kind.
+	if ref.Kind == "" {
+		name, version := catalog.ParseNameVersion(opts.Reference)
+		inferredKind, inferErr := catalog.InferKindFromRemote(ctx, remoteCatalog, name, version)
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/catalog/delete.go]: Fixed — the remote inference block now uses already-parsed `ref.Name` and `ref.Version` when available (full remote ref path), falling back to `ParseNameVersion(opts.Reference)` only for short refs where `ref` is zero-value. This prevents passing URL-like names into `InferKindFromRemote`. Thanks!

File: pkg/cmd/scafctl/build/solution.go [hunk 3]
```diff
@@ -260,6 +276,35 @@ func runBuildSolution(ctx context.Context, opts *SolutionOptions) error {
...
+	if !opts.DryRun {
+		reg, regErr := builtin.DefaultRegistry(ctx)
+		if regErr != nil {
+			reg = provider.GetGlobalRegistry()
+		}
+
+		pfResult, pfErr := builder.RunPreflight(ctx, &sol, opts.File, builder.PreflightOptions{
+			SkipLint:        opts.SkipLint,
+			SkipTests:       opts.SkipTests,
+			IgnorePreflight: opts.IgnorePreflight,
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go:302]: Fixed — `BinaryPath` is now set via `os.Executable()` before calling `RunPreflight`. Used `_` for the error since a missing executable is non-fatal (tests will simply be skipped if the path is empty). Thanks for the suggestion!

File: pkg/cmd/scafctl/build/solution.go [hunk 4]
```diff
@@ -195,6 +200,10 @@ func CommandBuildSolution(cliParams *settings.Run, ioStreams *terminal.IOStreams
 	cmd.Flags().BoolVar(&options.Dedupe, "dedupe", true, "Enable content-addressable deduplication")
 	cmd.Flags().StringVar(&options.DedupeThreshold, "dedupe-threshold", "4KB", "Minimum file size for individual layer extraction (smaller files are tarred together)")
 	cmd.Flags().BoolVar(&options.NoCache, "no-cache", false, "Skip build cache and force a full rebuild")
+	cmd.Flags().BoolVar(&options.SkipLint, "skip-lint", false, "Skip lint pre-flight check")
+	cmd.Flags().BoolVar(&options.SkipTests, "skip-tests", false, "Skip functional test pre-flight check")
+	cmd.Flags().BoolVar(&options.IgnorePreflight, "ignore-preflight", false, "Run pre-flight checks but proceed even if they fail")
+	cmd.Flags().BoolVar(&options.AllowDevVersion, "allow-dev-version", false, "Allow build without metadata.version set")
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go:206]: Fixed — added the 4 new flags (`skip-lint`, `skip-tests`, `ignore-preflight`, `allow-dev-version`) to the `TestCommandBuildSolution_Flags` table in `solution_coverage_test.go`. Good catch.

File: pkg/solution/builder/preflight.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		}
+
+		if tests != nil && len(tests.Cases) > 0 {
+			ios, _, _ := terminal.NewTestIOStreams()
+			runner := &soltesting.Runner{
+				BinaryPath:  opts.BinaryPath,
+				Concurrency: 1,
+				FailFast:    true,
+				IOStreams:   ios,
+			}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/solution/builder/preflight.go:132]: Fixed — replaced `terminal.NewTestIOStreams()` with `terminal.NewIOStreams(io.NopCloser(...), io.Discard, io.Discard, false)` so preflight test output goes to discard instead of growing in-memory buffers. Good catch.

File: pkg/catalog/url.go [hunk 6]
```diff
@@ -102,6 +102,46 @@ func InferKindFromLocalCatalog(ctx context.Context, localCatalog *LocalCatalog,
 	return "", fmt.Errorf("artifact %q not found in local catalog", name)
 }
 
+// InferKindFromRemote resolves an artifact's kind by trying each known kind
+// against the remote catalog. For each kind it attempts to resolve the tag;
+// the first successful resolve wins. This avoids fetching the full manifest
+// and works even when the artifact has no scafctl-specific annotations.
+//
+// Falls back to ArtifactKindSolution if no kind matches but the caller should
+// treat this as a best-effort guess.
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/url.go]: Fixed — updated the doc comment to match the actual behavior: returns an error when no kind matches, no fallback. Thanks for the suggestion.

File: pkg/catalog/remote.go [hunk 7]
```diff
@@ -633,6 +639,26 @@ func (c *RemoteCatalog) Delete(ctx context.Context, ref Reference) error {
...
+			if deleteErr := c.deleteByTag(ctx, ref, tag); deleteErr != nil {
+				return fmt.Errorf("failed to delete artifact by tag after digest rejection: %w", deleteErr)
+			}
+
+			c.logger.V(1).Info("deleted artifact via tag reference",
+				"name", ref.Name,
+				"tag", tag)
+
+			return nil
+		}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/remote.go:661]: This was addressed in a prior commit — the code no longer has the defer-in-loop issue. The comment is outdated.

File: pkg/catalog/remote.go [hunk 8]
```diff
@@ -643,6 +669,54 @@ func (c *RemoteCatalog) Delete(ctx context.Context, ref Reference) error {
...
+
+		status := resp.StatusCode
+		resp.Body.Close()
+
+		if status == http.StatusAccepted || status == http.StatusOK {
+			return nil
+		}
+		return fmt.Errorf("delete by tag returned status %d", status)
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/remote.go:717]: Good catch — added `http.StatusNoContent` (204) to the success check alongside 200 and 202. Thanks!

File: pkg/cmd/scafctl/catalog/delete.go [hunk 9]
```diff
@@ -249,6 +249,24 @@ func runDeleteRemote(ctx context.Context, opts *DeleteOptions) error {
...
+		if inferErr != nil {
+			w.Errorf("could not infer artifact kind: %v", inferErr)
+			w.Infof("Hint: use --kind to specify the artifact kind explicitly")
+			return exitcode.WithCode(inferErr, exitcode.InvalidInput)
+		}
+		ref, err = catalog.ParseReference(inferredKind, opts.Reference)
+		if err != nil {
+			w.Errorf("invalid reference: %v", err)
+			return exitcode.WithCode(err, exitcode.InvalidInput)
+		}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/catalog/delete.go:277]: Fixed — same as the other thread. The remote inference block now checks `ref.Name` first (set by the full-remote-ref path) and only falls back to `ParseNameVersion(opts.Reference)` for short refs where `ref` is zero-value. Thanks!

File: pkg/solution/builder/preflight.go [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+		if warnings == 0 {
+			warnings = countBySeverity(lintResult.Findings, lint.SeverityWarning)
+		}
+
+		switch {
+		case lintErrors > 0:
+			msg := fmt.Sprintf("lint: %d error(s), %d warning(s)", lintErrors, warnings)
+			if opts.IgnorePreflight {
+				result.Messages = append(result.Messages, msg+" (ignored via --ignore-preflight)")
+			} else {
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/solution/builder/preflight.go:95]: Good point — updated the messages to use generic phrasing (`"ignored, preflight errors overridden"`) instead of CLI-specific flag literals. This keeps the domain layer surface-agnostic. Thanks!

File: pkg/solution/builder/preflight.go [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+		opts.Logger.V(1).Info("running pre-flight functional tests")
+
+		tests, err := soltesting.DiscoverFromFile(filePath)
+		if err != nil {
+			return nil, fmt.Errorf("pre-flight test discovery: %w", err)
+		}
+
+		if tests != nil && len(tests.Cases) > 0 {
+			ios := terminal.NewIOStreams(io.NopCloser(io.LimitReader(nil, 0)), io.Discard, io.Discard, false)
+			runner := &soltesting.Runner{
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/solution/builder/preflight.go:127]: Fixed — replaced `io.LimitReader(nil, 0)` with `bytes.NewReader(nil)` which is a safe empty reader that returns EOF without risk of nil-pointer panic.

File: pkg/solution/builder/preflight.go [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+// parsed solution. Returns a result indicating whether the build should proceed.
+func RunPreflight(ctx context.Context, sol *solution.Solution, filePath string, opts PreflightOptions) (*PreflightResult, error) {
+	result := &PreflightResult{
+		TestsPassed: true,
+	}
+
+	// --- Lint ---
+	if !opts.SkipLint {
+		opts.Logger.V(1).Info("running pre-flight lint check")
+		lintResult := lint.Solution(sol, filePath, opts.Registry)
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/solution/builder/preflight.go:72]: Fixed — added `if opts.Logger.GetSink() == nil { opts.Logger = logr.Discard() }` at the top of RunPreflight, consistent with the pattern in pkg/httpc/filecache.go.

File: pkg/solution/builder/preflight.go [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+	// SkipLint skips the lint pre-flight check entirely.
+	SkipLint bool `json:"skipLint,omitempty" yaml:"skipLint,omitempty" doc:"Skip lint pre-flight check"`
+
+	// SkipTests skips the functional test pre-flight check entirely.
+	SkipTests bool `json:"skipTests,omitempty" yaml:"skipTests,omitempty" doc:"Skip functional test pre-flight check"`
+
+	// IgnorePreflight runs checks but treats errors as warnings instead of blocking.
+	IgnorePreflight bool `json:"ignorePreflight,omitempty" yaml:"ignorePreflight,omitempty" doc:"Run checks but treat errors as warnings"`
+
+	// BinaryPath is the scafctl binary path for test execution.
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/solution/builder/preflight.go]: Good catch — updated the comment to "CLI executable path used for test execution" to be embedder-neutral. Thanks!

File: pkg/cmd/scafctl/build/solution.go [hunk 14]
```diff
@@ -260,6 +276,38 @@ func runBuildSolution(ctx context.Context, opts *SolutionOptions) error {
...
 	}
 
+	// === Pre-flight checks ===
+	if !opts.DryRun {
+		reg, regErr := builtin.DefaultRegistry(ctx)
+		if regErr != nil {
+			reg = provider.GetGlobalRegistry()
+		}
+
+		binaryPath, _ := os.Executable()
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go]: Fixed — registry is now only initialized when lint will run, and os.Executable() only when tests will run. Accepted the suggestion. Thanks!

File: pkg/catalog/remote.go [hunk 15]
```diff
@@ -633,6 +639,26 @@ func (c *RemoteCatalog) Delete(ctx context.Context, ref Reference) error {
...
+		// GCP Artifact Registry rejects digest-based DELETE when the manifest
+		// still has a tag. Retry by deleting via the tag reference, which
+		// removes both the tag and the manifest in one operation.
+		if strings.Contains(errStr, "dangling tag") || strings.Contains(errStr, "still referenced") {
+			c.logger.V(1).Info("digest delete rejected (tagged manifest), retrying with tag reference",
+				"tag", tag, "digest", desc.Digest.String())
+
+			if deleteErr := c.deleteByTag(ctx, ref, tag); deleteErr != nil {
+				return fmt.Errorf("failed to delete artifact by tag after digest rejection: %w", deleteErr)
+			}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/remote.go:654]: Fixed — the tag-fallback now only triggers when `ref.Version != nil` (tag/version-based reference). For digest-based deletes, the original error is returned as-is since retrying with the same digest would be pointless.

File: pkg/catalog/remote.go [hunk 16]
```diff
@@ -633,6 +639,26 @@ func (c *RemoteCatalog) Delete(ctx context.Context, ref Reference) error {
...
+
+			c.logger.V(1).Info("deleted artifact via tag reference",
+				"name", ref.Name,
+				"tag", tag)
+
+			return nil
+		}
+
 		return fmt.Errorf("failed to delete artifact: %w", err)
 	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/remote.go:665]: Fixed — changed `ref.Version.String()` to `ref.VersionOrDigest()` so digest-based deletes are safe.

File: pkg/cmd/scafctl/build/solution.go [hunk 17]
```diff
@@ -239,6 +248,13 @@ func runBuildSolution(ctx context.Context, opts *SolutionOptions) error {
 		return exitcode.WithCode(err, exitcode.InvalidInput)
 	}
 
+	// Block dev version unless explicitly allowed
+	if version.Equal(solution.DefaultVersion()) && !opts.AllowDevVersion {
+		w.Errorf("metadata.version is not set (resolved to %s)", version.String())
+		w.Infof("Set metadata.version, pass --version, or use %s build solution --allow-dev-version to build anyway", opts.CliParams.BinaryName)
+		return exitcode.Errorf("dev version not allowed")
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go:260]: Added `TestRunBuildSolution_NoVersion_AllowDevVersion` which verifies that a solution without metadata.version fails by default (existing test now asserts "dev version not allowed") and succeeds with `AllowDevVersion: true` + `DryRun: true`. Thanks for the suggestion!

File: pkg/cmd/scafctl/build/solution_coverage_test.go [hunk 18]
```diff
@@ -187,6 +191,171 @@ spec: {}
...
+		CliParams:     settings.NewCliParams(),
+		SkipLint:      true,
+		SkipTests:     true,
+		DryRun:        true,
+		BundleMaxSize: "50MB",
+	}
+
+	// With lint and tests skipped, preflight should pass and dry-run should succeed.
+	err := runBuildSolution(ctx, opts)
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution_coverage_test.go]: Fixed — removed `DryRun: true` so the test actually exercises the preflight path with `SkipLint`/`SkipTests`. Now asserts output contains `"lint: skipped"` and `"tests: skipped"` to verify preflight ran. Thanks!

File: pkg/cmd/scafctl/build/solution_coverage_test.go [hunk 19]
```diff
@@ -187,6 +191,171 @@ spec: {}
...
+		CliParams:       settings.NewCliParams(),
+		SkipTests:       true,
+		IgnorePreflight: true,
+		DryRun:          true,
+		BundleMaxSize:   "50MB",
+	}
+
+	// With IgnorePreflight, lint errors become warnings and build proceeds.
+	err := runBuildSolution(ctx, opts)
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution_coverage_test.go]: Fixed — same approach as the SkipLint test: removed `DryRun: true` so `IgnorePreflight` is actually exercised. Asserts output contains `"lint:"` and does NOT contain `"build blocked"`. Thanks!

File: pkg/catalog/url.go [hunk 20]
```diff
@@ -102,6 +102,46 @@ func InferKindFromLocalCatalog(ctx context.Context, localCatalog *LocalCatalog,
...
+		if err != nil {
+			continue
+		}
+		if exists {
+			return kind, nil
+		}
+	}
+
+	return "", fmt.Errorf("artifact %q not found in remote catalog", name)
+}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/url.go:148]: Fixed — `InferKindFromRemote` now collects the last error from `Exists()` and returns `"failed to query remote catalog for %q: %w"` wrapping the underlying error when all kinds fail with errors, distinguishing it from the `"artifact %q not found"` case. Note: `Exists()` currently swallows errors internally (`return err == nil, nil`), so this is future-proofing for when that method is updated to propagate non-not-found errors. Thanks!

File: pkg/catalog/remote.go [hunk 21]
```diff
@@ -633,16 +639,84 @@ func (c *RemoteCatalog) Delete(ctx context.Context, ref Reference) error {
...
+		}
+
+		status := resp.StatusCode
+		resp.Body.Close()
+
+		if status == http.StatusOK || status == http.StatusAccepted || status == http.StatusNoContent {
+			return nil
+		}
+		return fmt.Errorf("delete by tag returned status %d", status)
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/catalog/remote.go:717]: Fixed — now drains the response body with `io.Copy(io.Discard, resp.Body)` before closing, enabling HTTP connection reuse. Thanks!

File: issues/feat/auth/feat-auth-device-code-ui.md
Comment (kcloutie) [inline issues/feat/auth/feat-auth-device-code-ui.md:1]: Why not create a github issue with this content?

File: issues/feat/cli/feat-action-filter.md
Comment (kcloutie) [inline issues/feat/cli/feat-action-filter.md:1]: Why not create a github issue with this content?

File: issues/feat/cli/feat-exec-output-shorthand.md
Comment (kcloutie) [inline issues/feat/cli/feat-exec-output-shorthand.md:1]: Why not create a github issue with this content?

File: issues/feat/cli/feat-exec-passthrough.md
Comment (kcloutie) [inline issues/feat/cli/feat-exec-passthrough.md:1]: Why not create a github issue with this content?

File: issues/feat/cli/feat-interactive-help.md
Comment (kcloutie) [inline issues/feat/cli/feat-interactive-help.md:1]: Why not create a github issue with this content?

File: issues/feat/solution/feat-on-conflict-default-error.md
Comment (kcloutie) [inline issues/feat/solution/feat-on-conflict-default-error.md:1]: Why not create a github issue with this content?

File: issues/feat/solution/feat-solution-inspect.md
Comment (kcloutie) [inline issues/feat/solution/feat-solution-inspect.md:1]: Why not create a github issue with this content?

File: issues/feat/solution/feat-unified-solution-discovery.md
Comment (kcloutie) [inline issues/feat/solution/feat-unified-solution-discovery.md:1]: Why not create a github issue with this content?

File: issues/fix/fix-build-preflight-checks.md
Comment (kcloutie) [inline issues/fix/fix-build-preflight-checks.md:1]: Why not create a github issue with this content?

File: issues/fix/fix-catalog-remote-delete.md
Comment (kcloutie) [inline issues/fix/fix-catalog-remote-delete.md:1]: Why not create a github issue with this content?

File: issues/question/feat-valueref-expansion.md
Comment (kcloutie) [inline issues/question/feat-valueref-expansion.md:1]: Why not create a github issue with this content?

File: issues/refactor/refactor-consistent-kvx-output.md
Comment (kcloutie) [inline issues/refactor/refactor-consistent-kvx-output.md:1]: Why not create a github issue with this content?

File: pkg/catalog/remote.go
Comment (kcloutie) [inline pkg/catalog/remote.go:1]: Why not create a github issue with this content?

**Review by kcloutie — CHANGES_REQUESTED** [review]
Some questions

File: pkg/cmd/scafctl/build/solution.go [hunk 22]
```diff
@@ -260,6 +276,45 @@ func runBuildSolution(ctx context.Context, opts *SolutionOptions) error {
 		}
 	}
 
+	// === Pre-flight checks ===
+	if !opts.DryRun {
+		var reg *provider.Registry
+		if !opts.SkipLint {
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go:286]: Added `TestRunBuildSolution_DryRunSkipsPreflight` — asserts no `"lint:"` or `"tests:"` messages appear in output when `DryRun=true`, and confirms `"Dry run:"` is present. Thanks!

File: pkg/catalog/remote.go
Comment (abaker-9) [reply pkg/catalog/remote.go:1]: The issue content has been moved out of this PR (issues folder moved to temp/). It will be filed as separate GitHub issues. Thanks for the catch!

File: pkg/cmd/scafctl/build/solution.go [hunk 23]
```diff
@@ -239,6 +248,13 @@ func runBuildSolution(ctx context.Context, opts *SolutionOptions) error {
 		return exitcode.WithCode(err, exitcode.InvalidInput)
 	}
 
+	// Block dev version unless explicitly allowed
+	if version.Equal(solution.DefaultVersion()) && !opts.AllowDevVersion {
+		w.Errorf("metadata.version is not set (resolved to %s)", version.String())
+		w.Infof("Set metadata.version, pass --version, or use %s build solution --allow-dev-version to build anyway", opts.CliParams.BinaryName)
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution.go]: Fixed — added nil guard using `settings.BinaryNameFromContext(ctx)` as the default, with `opts.CliParams.BinaryName` override when available. Matches the suggested code exactly. Thanks!

File: pkg/cmd/scafctl/build/solution_coverage_test.go [hunk 24]
```diff
@@ -187,6 +191,175 @@ spec: {}
...
+		SkipLint:      true,
+		SkipTests:     true,
+		BundleMaxSize: "50MB",
+	}
+
+	// With lint and tests skipped, preflight runs but skips both checks.
+	// Subsequent pipeline steps may fail; we only care about the preflight path.
+	_ = runBuildSolution(ctx, opts)
+	output := buf.String()
+	assert.Contains(t, output, "lint: skipped")
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution_coverage_test.go:298]: Fixed — added `t.Setenv("XDG_DATA_HOME", tmpDir)` / `t.Setenv("XDG_CACHE_HOME", tmpDir)` + `xdg.Reload()` with cleanup. Removed `t.Parallel()` since `t.Setenv` is incompatible with parallel tests. Thanks!

File: pkg/cmd/scafctl/build/solution_coverage_test.go [hunk 25]
```diff
@@ -187,6 +191,175 @@ spec: {}
...
+		IgnorePreflight: true,
+		BundleMaxSize:   "50MB",
+	}
+
+	// With IgnorePreflight, lint errors become warnings and the build is not blocked.
+	// Subsequent pipeline steps may fail; we only care about the preflight path.
+	_ = runBuildSolution(ctx, opts)
+	output := buf.String()
+	assert.Contains(t, output, "lint:")
+	assert.NotContains(t, output, "build blocked")
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/build/solution_coverage_test.go:339]: Fixed — same XDG isolation applied here. Thanks!
